assets/js/googlesitekit/modules/create-submit-changes-store.js

```diff
@@ -0,0 +1,201 @@
+/**
+ * API function to create submit changes store.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import API from 'googlesitekit-api';
+import { TYPE_MODULES } from '../../components/data/constants';
+import { invalidateCacheGroup } from '../../components/data/invalidate-cache-group';
+import { createStateSelectors, createStrictSelect, createValidationSelector } from '../data/utils';
+const { createRegistryControl } = Data;
+
+// Invariant error messages.
+export const INVARIANT_DOING_SUBMIT_CHANGES = 'cannot submit changes while submitting changes';
+export const INVARIANT_SETTINGS_NOT_CHANGED = 'cannot submit changes if settings have not changed';
+
+// Actions
+const SUBMIT_CHANGES = 'SUBMIT_CHANGES';
+const START_SUBMIT_CHANGES = 'START_SUBMIT_CHANGES';
+const FINISH_SUBMIT_CHANGES = 'FINISH_SUBMIT_CHANGES';
+
+/**
+ * Creates a store object implementing the necessary infrastructure for submitting module settings.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Object}   args                            Arguments for creating the submitChanges store.
+ * @param {string}   args.storeName                  Datastore slug.
+ * @param {Function} [args.submitChanges]            Optional. Callback function to issue the submit changes request. Will be used inside the submit changes control.
+ * @param {Function} [args.validateCanSubmitChanges] Optional. A helper function to validate that settings can be submitted.
+ * @return {Object} Partial store object with properties 'actions', 'controls', 'reducer', 'resolvers', and 'selectors'.
+ */
+export function createSubmitChangesStore( {
+	storeName,
+	submitChanges = () => () => ( {} ),
+	validateCanSubmitChanges = () => {},
+} = {} ) {
+	invariant( storeName, 'storeName is required.' );
+
+	const initialState = {
+		hasStartedSubmittingChanges: false,
+		hasFinishedSubmittingChanges: false,
```

Comment:
Let's remove these two - we only need `isDoingSubmitChanges` like in the existing implementations.